### PR TITLE
Add escape sequence test and ensure .trim() lookup rules are applied to in-mem filtering logic

### PR DESCRIPTION
**Pull request by iambmelt**
Adds a test to demonstrate that escape sequences are trimmed from cache keys

### CODE REVIEW

File: common/src/test/java/com/microsoft/identity/common/CacheKeyValueDelegateTest.java [hunk 1]
```diff
@@ -95,6 +95,26 @@ public void accessTokenCreateCacheKeyComplete() {
         assertEquals(expectedKey, mDelegate.generateCacheKey(accessToken));
     }
 
+    @Test
+    public void accessTokenCreateCacheKeyCompleteWithEscapeSequences() {
+        final AccessTokenRecord accessToken = new AccessTokenRecord();
+        accessToken.setHomeAccountId(HOME_ACCOUNT_ID);
+        accessToken.setEnvironment("\r\f\n\t" + ENVIRONMENT + "\r\f\n\t");
```
Comment (AdamBJohnsonx) [inline common/src/test/java/com/microsoft/identity/common/CacheKeyValueDelegateTest.java:102]: Is this the only field that has this behavior?

Comment (iambmelt) [reply common/src/test/java/com/microsoft/identity/common/CacheKeyValueDelegateTest.java:102]: They all should - can broaden this test to make this clearer.

**Review by AdamBJohnsonx — APPROVED** [review]
ship it.  To the point, here, have we put a breakpoint in the java code and then managed to hit it with the C++ test to see what we're actually getting sent?

**Comment by iambmelt** [comment]
> ship it. To the point, here, have we put a breakpoint in the java code and then managed to hit it with the C++ test to see what we're actually getting sent?

Yep - it looks like the in-mem filtering is where this went awry, so shipped a simple fix in 699fa0ace59f9ef7ba4fc4cecc78c12b4729315d to make it behave the same as would a key-based lookup.

**Pull request merged**